Mathlib/Data/Int/WithZero.lean

```diff
@@ -42,8 +42,7 @@ def toNNReal {e : ℝ≥0} (he : e ≠ 0) : ℤₘ₀ →*₀ ℝ≥0 where
   toFun := fun x ↦ if hx : x = 0 then 0 else e ^ (WithZero.unzero hx).toAdd
   map_zero' := rfl
   map_one' := by
-    simp only [dif_neg one_ne_zero]
-    erw [toAdd_one, zpow_zero]
+    simp [One.eq_ofNat1, dif_neg one_ne_zero]
```

Comment:
I think the actual problem here is that `unzero_coe` is somehow misfiring on the numeral

Reply:
Probably the thing to do is add a new simp lemma `unzero_one`  instead.

Reply:
I don't think we should be adding any lemmas that mention `One.one`. Can you add a `unzero_one` lemma instead, which ensures that the `One.one` never appears in the first place?

Reply:
I guess I can do that, but I have to make it higher priority for it to trigger rather than `unzero_coe`. 

I am nervous that I am leaving a problem unsolved. Do you have ideas why applying `unzero_coe` causes this `1` to convert to a `one` in the first place?

Reply:
I think it's because the argument to unzero is implicit. If you change it to be explicit (which is also good for the goal view), then I think the lemma will stop applying.